play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSRequestSpec.scala

```diff
@@ -152,22 +163,30 @@ class AhcWSRequestSpec extends Specification with Mockito with DefaultBodyReadab
       }
     }
 
+    "set content type explicitly if its not set with a body" in {
+      val client = mock[StandaloneAhcWSClient]
+      val request = new StandaloneAhcWSRequest(client, "http://example.com", /*materializer*/ null)
+      request.setBody(body("HELLO WORLD", null)) // content type is not set
+      request.addHeader("Content-Type", "application/json") // will be used as content type is not set with a body
+      val req = request.buildRequest()
+      req.getHeaders.get("Content-Type") must be_==("application/json")
+    }
+
     "only send first content type header and add charset=utf-8 to the Content-Type header if it's manually adding but lacking charset" in {
```

Comment:
Can you update the description to match what you are testing here? It should be something like "not set Content-Type header if the body does not set it" and same pattern above "set Content-Type header if the body sets it".